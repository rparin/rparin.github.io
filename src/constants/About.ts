const AboutGen = {
  Initials: "RP",
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, Web Developer, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels",
  "playing games",
  "watching movies",
  "learning new things",
];

const Socials = {
  github: "https://github.com/rparin",
  linkedIn: "https://www.linkedin.com/in/rparin",
};

const AboutMe = {
  Intro: "A little about me: my hobbies, interests, and goals.",

  Goals:
    "The thrill of crafting with my hands led me to computer science, where I aspire to make a meaningful difference in the lives of others, whether through a useful digital tool or a fun game.",
  GoalsSm:
    "Crafting drove me to computer science to make a difference, via games or digital tools.",

  Hobbies:
    "I cherish stories for the unique lessons they offer, particularly the no-happy-ending tales by my favorite author, Sugaru Miaki. They've taught me to embrace failure as a path to improvement.",
  HobbiesSm:
    "I enjoy reading for the lessons they offer. They've, taught me to overcome my regrets.",

  Interests:
    "One of my passion lies in creating memorable moments with friendsâ€”whether it's trying new flavors at local eateries, embarking on food adventures, or exploring exciting new places together.",
  InterestsSm:
    "One hobby of mine includes trying new foods and sightseeing with friends.",

  Gaming:
    "Gaming with my friends has become my digital backyard, as life takes us on different paths and distances grow, this space keeps us connected.",
  GamingSm:
    "Gaming with friends is my digital backyard, keeping me close despite life's changes.",
};

export { AboutGen, AboutSent, Socials, AboutMe };
