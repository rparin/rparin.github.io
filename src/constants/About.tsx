const AboutGen = {
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, WebDeveloper, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels.",
  "playing games.",
  "watching movies.",
  "trying new things.",
];

export { AboutGen, AboutSent };
