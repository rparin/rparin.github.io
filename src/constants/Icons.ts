import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";
import zod from "@icons/Zod.svg";
import springboot from "@icons/SpringBoot.svg";
import docker from "@icons/Docker.svg";

type Icon = {
  name: string;
  img: StaticImageData;
  alt: string;
  invert?: boolean;
  url?: string;
};

const SocialIcons = {
  github: {
    name: "Github",
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    name: "LinkedIn",
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    name: "Gmail",
    img: gmail,
    alt: "logo of Gmail",
  },
};

const DevTools = {
  angular: {
    name: "Angular",
    img: angular,
    alt: "logo of angular",
    url: "https://angular.io/",
  },
  react: {
    name: "React",
    img: react,
    alt: "logo of react",
    url: "https://react.dev/",
  },
  nextjs: {
    name: "Nextjs",
    img: nextjs,
    alt: "logo of nextjs",
    url: "https://nextjs.org/",
  },
  springboot: {
    name: "Spring Boot",
    img: springboot,
    alt: "logo of Spring Boot",
    url: "https://spring.io/projects/spring-boot",
  },
  reactquery: {
    name: "Tanstack Query",
    img: reactquery,
    alt: "logo of reactquery",
    url: "https://tanstack.com/query/latest",
  },
  zod: {
    name: "Zod",
    img: zod,
    alt: "logo of zod",
    url: "https://zod.dev",
  },
  docker: {
    name: "Docker",
    img: docker,
    alt: "logo of Docker",
    url: "https://www.docker.com",
  },
  mapbox: {
    name: "Mapbox",
    img: mapbox,
    alt: "logo of mapbox",
    url: "https://www.mapbox.com/",
  },
  ionic: {
    name: "Ionic",
    img: ionic,
    alt: "logo of ionic",
    url: "https://ionicframework.com/",
  },
  tailwind: {
    name: "Tailwind",
    img: tailwind,
    alt: "logo of tailwind",
    url: "https://tailwindcss.com/",
  },
  fbase: {
    name: "Firebase",
    img: fbase,
    alt: "logo of firebase",
    url: "https://firebase.google.com/",
  },
  nodejs: {
    name: "Nodejs",
    img: nodejs,
    alt: "logo of nodejs",
    url: "https://nodejs.org/en",
  },
  cpp: {
    name: "C++",
    img: cpp,
    alt: "logo of c++",
    url: "https://isocpp.org/",
  },
  css: {
    name: "CSS",
    img: css,
    alt: "logo of css",
    url: "https://developer.mozilla.org/en-US/docs/Web/CSS",
  },
  js: {
    name: "Javascript",
    img: js,
    alt: "logo of javascript",
    url: "https://developer.mozilla.org/en-US/docs/Web/javascript",
  },
  html: {
    name: "HTML",
    img: html,
    alt: "logo of html",
    url: "https://developer.mozilla.org/en-US/docs/Web/HTML",
  },
  ts: {
    name: "Typescript",
    img: ts,
    alt: "logo of typescript",
    url: "https://www.typescriptlang.org/",
  },
  python: {
    name: "Python",
    img: python,
    alt: "logo of python programming language",
    url: "https://www.python.org/",
  },
  java: {
    name: "Java",
    img: java,
    alt: "logo of java",
    url: "https://www.java.com/en/download/help/whatis_java.html",
  },
  pygame: {
    name: "Pygame",
    img: pygame,
    alt: "logo of pygame library",
    url: "https://www.pygame.org/",
  },
  sfml: {
    name: "SFML",
    img: sfml,
    alt: "logo of SFML library",
    url: "https://www.sfml-dev.org/",
  },
  tensflow: {
    name: "Tensorflow",
    img: tensflow,
    alt: "logo of tensorflow",
    url: "https://www.tensorflow.org/",
  },
  pandas: {
    name: "Pandas",
    img: pandas,
    alt: "logo of pandas library",
    url: "https://pandas.pydata.org/",
  },
  flask: {
    name: "Flask",
    img: flask,
    alt: "logo of flask framework",
    url: "https://flask.palletsprojects.com/en/3.0.x/",
  },
  openai: {
    name: "OpenAi",
    img: openai,
    alt: "logo of openai",
    url: "https://openai.com/",
  },
};

export { DevTools, SocialIcons };
export type { Icon };
