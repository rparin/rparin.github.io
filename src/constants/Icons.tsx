import { StaticImageData } from "next/image";
import git from "@icons/Github.svg";
import linked from "@icons/LinkedIn.svg";
import gmail from "@icons/GMail.svg";
import download from "@icons/Download.svg";
import cpp from "@icons/Cpp.svg";
import css from "@icons/CSS.svg";
import js from "@icons/JS.svg";
import html from "@icons/HTML.svg";
import java from "@icons/Java.svg";
import ts from "@icons/TS.svg";
import angular from "@icons/Angular.svg";
import react from "@icons/React.svg";
import nodejs from "@icons/NodeJs.svg";
import fbase from "@icons/Firebase.svg";
import python from "@icons/Python.svg";
import ionic from "@icons/Ionic.svg";
import tailwind from "@icons/Tailwind.svg";
import pygame from "@icons/pygame.svg";
import sfml from "@icons/SFML.svg";
import openai from "@icons/OpenAi.svg";
import pandas from "@icons/Pandas.svg";
import flask from "@icons/Flask.svg";
import tensflow from "@icons/Tensorflow.svg";
import mapbox from "@icons/mapbox.svg";
import nextjs from "@icons/nextjs.svg";
import reactquery from "@icons/reactquery.svg";

type Img = {
  img: StaticImageData;
  alt: string;
  invert?: boolean;
};

const MiscIcons = {
  download: {
    img: download,
    alt: "download icon",
  },
};

const SocialIcons = {
  github: {
    img: git,
    alt: "logo of Github",
  },
  linkedin: {
    img: linked,
    alt: "logo of LinkedIn",
  },
  gmail: {
    img: gmail,
    alt: "logo of Gmail",
  },
};

const CodeIcons = {
  angular: {
    img: angular,
    alt: "logo of angular",
  },
  react: {
    img: react,
    alt: "logo of react",
  },
  nextjs: {
    img: nextjs,
    alt: "logo of nextjs",
  },
  reactquery: {
    img: reactquery,
    alt: "logo of reactquery",
  },
  mapbox: {
    img: mapbox,
    alt: "logo of mapbox",
  },
  ionic: {
    img: ionic,
    alt: "logo of ionic",
  },
  tailwind: {
    img: tailwind,
    alt: "logo of tailwind",
  },
  fbase: {
    img: fbase,
    alt: "logo of firebase",
  },
  nodejs: {
    img: nodejs,
    alt: "logo of nodejs",
  },
  cpp: {
    img: cpp,
    alt: "logo of c++",
  },
  css: {
    img: css,
    alt: "logo of css",
  },
  js: {
    img: js,
    alt: "logo of javascript",
  },
  html: {
    img: html,
    alt: "logo of html",
  },
  ts: {
    img: ts,
    alt: "logo of typescript",
  },
  python: {
    img: python,
    alt: "logo of python programming language",
  },
  java: {
    img: java,
    alt: "logo of java",
  },
  pygame: {
    img: pygame,
    alt: "logo of pygame library",
  },
  sfml: {
    img: sfml,
    alt: "logo of SFML library",
  },
  tensflow: {
    img: tensflow,
    alt: "logo of tensorflow",
  },
  pandas: {
    img: pandas,
    alt: "logo of pandas library",
  },
  flask: {
    img: flask,
    alt: "logo of flask framework",
  },
  openai: {
    img: openai,
    alt: "logo of openai",
    invert: true,
  },
};

export { MiscIcons, CodeIcons, SocialIcons };
export type { Img };
