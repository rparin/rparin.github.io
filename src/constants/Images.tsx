import landBgR from "../assets/Landing/Red.svg";
import landBgP from "../assets/Landing/Purple.svg";
import landCode from "../assets/Landing/CodePy.webp";
import AboutCode from "../assets/About/CodeE.webp";
import AboutFood from "../assets/About/FoodE.webp";
import AboutGame from "../assets/About/LaptopE.webp";
import AboutBook from "../assets/About/BookE.webp";

const HomeImages = {
  bgR: {
    img: landBgR,
    alt: "Landing page background",
  },
  bgP: {
    img: landBgP,
    alt: "Landing page background",
  },
  code: {
    img: landCode,
    alt: "Landing page code background overlay",
  },
};

const AboutImages = {
  code: {
    img: AboutCode,
    alt: "Background image of study room",
  },
  book: {
    img: AboutBook,
    alt: "Background image of book in forest",
  },
  food: {
    img: AboutFood,
    alt: "Background image of food in cafe",
  },
  game: {
    img: AboutGame,
    alt: "Background image of laptop in urban city",
  },
};

export { HomeImages, AboutImages };
