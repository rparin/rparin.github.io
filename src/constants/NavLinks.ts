const NavItems = {
  projects: {
    id: "projects",
    title: "Projects",
    url: undefined,
  },
  about: {
    id: "about",
    title: "About",
    url: undefined,
  },
  contact: {
    id: "contact",
    title: "Contact",
    url: undefined,
  },
};

const NavHome = {
  id: "home",
};

const Resume = {
  id: "resume",
  title: "Resume",
  url: "/Ralph_Parin_Resume.pdf",
};

export { NavItems, NavHome, Resume };
