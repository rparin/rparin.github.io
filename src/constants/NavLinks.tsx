const NavItems = {
  projects: {
    id: "projects",
    title: "Projects",
  },
  about: {
    id: "about",
    title: "About",
  },
  contact: {
    id: "contact",
    title: "Contact",
  },
  resume: {
    id: "resume",
    title: "Resume",
    url: "https://raw.githubusercontent.com/rparin/rparin.github.io/main/public/Resume.pdf",
  },
};

const NavLinks = [
  {
    id: NavItems.projects.id,
    title: NavItems.projects.title,
  },
  {
    id: NavItems.about.id,
    title: NavItems.about.title,
  },
  {
    id: NavItems.contact.id,
    title: NavItems.contact.title,
  },
  {
    id: NavItems.resume.id,
    title: NavItems.resume.title,
    url: NavItems.resume.url,
  },
];

export { NavItems, NavLinks };
