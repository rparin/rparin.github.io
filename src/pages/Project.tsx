import { ProjectItems } from "@/constants/Projects";
import Image from "next/image";

export default function Project() {
  const PItems = [];
  for (const [k, project] of Object.entries(ProjectItems)) {
    PItems.push(
      <Image
        key={k}
        className={project.span ? "col-span-2" : undefined}
        src={project.img}
        alt={project.alt}
      />
    );
  }

  return (
    <section>
      <div className="mt-5 grid auto-cols-auto md:grid-cols-4 xl:grid-cols-5 gap-3 mx-16">
        {PItems}
      </div>
    </section>
  );
}
