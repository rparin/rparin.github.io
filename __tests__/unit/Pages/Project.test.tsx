import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import { NavItems } from "@/constants/NavLinks";
import Project from "@/pages/Project";

describe("testing Project Page Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(<Project id={NavItems.projects.id} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<Project id={NavItems.projects.id} />);
  expect(container).not.toMatchSnapshot();
});
