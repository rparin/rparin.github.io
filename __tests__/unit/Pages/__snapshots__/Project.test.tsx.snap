// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <section
    class="scroll-mt-[70px]"
    id="projects"
  >
    <h2
      class="sr-only"
    >
      Projects
    </h2>
    <ul
      aria-label="Projects"
      class="mx-16 mt-5 grid auto-cols-auto gap-2 md:grid-cols-4 xl:grid-cols-5"
    >
      <li
        class="group relative col-span-2 md:hover:overflow-y-auto"
        tabindex="0"
      >
        <div
          aria-label="Spotify Artist Network"
          class="from-1% absolute hidden max-h-full w-auto grid-cols-4 grid-rows-4 bg-gradient-to-t from-background text-base font-semibold text-text backdrop-blur-md group-focus-within:grid group-hover:grid md:grid-cols-5 md:grid-rows-5"
          role="grid"
        >
          <img
            alt="Demo of Spotify Artist Network"
            aria-hidden="true"
            class="col-span-3 row-span-3 md:col-span-4 md:row-span-4"
            data-nimg="1"
            decoding="async"
            height="924"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FDemo.gif&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FDemo.gif&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FDemo.gif&w=3840&q=75 2x"
            style="color: transparent;"
            width="1912"
          />
          <div
            aria-label="Dev Tools"
            class="p-scroll row-span-3 mx-1 mt-1 flex flex-col items-center justify-start gap-2 overflow-y-auto md:row-span-4 lg:mt-2"
            role="row"
          >
            <span
              role="cell"
            >
              <a
                aria-label="React"
                href="https://react.dev/"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of react"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Nodejs"
                href="https://nodejs.org/en"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of nodejs"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Nextjs"
                href="https://nextjs.org/"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of nextjs"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Zod"
                href="https://zod.dev"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of zod"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Tanstack Query"
                href="https://tanstack.com/query/latest"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of reactquery"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Tailwind"
                href="https://tailwindcss.com/"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of tailwind"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Typescript"
                href="https://www.typescriptlang.org/"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of typescript"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="HTML"
                href="https://developer.mozilla.org/en-US/docs/Web/HTML"
                rel="noopener noreferrer"
                tabindex="-1"
                target="_blank"
              >
                <img
                  alt="logo of html"
                  class="h-auto w-7 md:w-10 lg:w-14 md:p-1"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
          </div>
          <div
            class="col-span-4 flex flex-row items-center justify-center gap-1 truncate md:col-span-5"
            role="row"
          >
            <span
              role="rowheader"
            >
              Spotify Artist Network
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Website"
                href="https://spotify-artist-network.vercel.app"
                rel="noopener noreferrer"
                target="_blank"
              >
                <svg
                  class="mb-2 h-3 w-auto md:h-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
                  />
                  <polyline
                    points="15 3 21 3 21 9"
                  />
                  <line
                    x1="10"
                    x2="21"
                    y1="14"
                    y2="3"
                  />
                </svg>
              </a>
            </span>
            <span
              role="cell"
            >
              <a
                aria-label="Github"
                href="https://github.com/rparin/SpotifyArtistNetwork"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="logo of Github"
                  aria-hidden="true"
                  class="ml-1 h-5 w-auto dark:invert md:h-7"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </a>
            </span>
          </div>
        </div>
        <img
          alt="Spotify Artist Network Landing page screenshot"
          aria-hidden="true"
          data-nimg="1"
          decoding="async"
          height="924"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FSpotifyArtistNetwork.webp&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FSpotifyArtistNetwork.webp&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fraw.githubusercontent.com%2Frparin%2FSpotifyArtistNetwork%2Fmain%2F_preview%2FSpotifyArtistNetwork.webp&w=3840&q=75 2x"
          style="color: transparent;"
          width="1912"
        />
      </li>
    </ul>
  </section>
</div>
`;
